BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -234,48 +225,66 @@ public final class ConvivaAnalytics: NSObject {
      */
     public func pauseTracking() {
         // AdStart is the preferred way to pause tracking according to conviva.
-        client.adStart(sessionKey,
-                       adStream: .ADSTREAM_SEPARATE,
-                       adPlayer: .ADPLAYER_SEPARATE,
-                       adPosition: .ADPOSITION_PREROLL)
-        client.detachPlayer(sessionKey)
+        videoAnalytics?.reportAdBreakStarted(AdPlayer.ADPLAYER_SEPARATE,
+                                             adType: AdTechnology.CLIENT_SIDE,
+                                             adBreakInfo: [:])
         logger.debugLog(message: "Tracking paused.")
     }
 
     /**
      Puts the session state from a notMonitored state into the last one tracked.
      */
     public func resumeTracking() {
-        client.attachPlayer(sessionKey, playerStateManager: playerStateManager)
         // AdEnd is the preferred way to resume tracking according to conviva.
-        client.adEnd(sessionKey)
+        videoAnalytics?.reportAdBreakEnded()
         logger.debugLog(message: "Tracking resumed.")
     }
 
     // MARK: - session handling
-    private func setupPlayerStateManager() {
-        playerStateManager = client.getPlayerStateManager()
-        playerStateManager.setPlayerState!(PlayerState.CONVIVA_STOPPED)
-        playerStateManager.setPlayerType!("Bitmovin Player iOS")
+
+    private func setupPlayerInfo() {
+        videoAnalytics?.reportPlaybackMetric(CIS_SSDK_PLAYBACK_METRIC_PLAYER_STATE,
+                                             value: PlayerState.CONVIVA_STOPPED.rawValue)
+
+        var playerInfo = [String: Any]()
+        playerInfo[CIS_SSDK_PLAYER_FRAMEWORK_NAME] = "Bitmovin Player iOS"
 
         if let bitmovinPlayerVersion = playerHelper.version {
-            playerStateManager.setPlayerVersion!(bitmovinPlayerVersion)
+            playerInfo[CIS_SSDK_PLAYER_FRAMEWORK_VERSION] = bitmovinPlayerVersion
         }
+        videoAnalytics?.setPlayerInfo(playerInfo)
     }
 
-    private func internalInitializeSession() {
+    private func internalInitializeSession(fromPlay: Bool) {
         buildContentMetadata()
 
-        sessionKey = client.createSession(with: contentMetadataBuilder.build())
+        if fromPlay {
```

Comment:
Why do we need the `fromPlay` at all 🤔 Can't we just always initialize when not yet initialized? The `isSessionActive` should already handle that or? 🤔

Reply:
Great point. Initing when the session is not active here may work. It seems like that was not applicable to the previous API.
Though it is not 100% clear when we would get into this state exactly. It's possible we would not always want to init here? Possible if we get events while we are deinitializing the player?

Reply:
> It's possible we would not always want to init here?

Yes, definitely (we had an issue when a user seeks before starting playback in the past). IIRC we have to init the session on the following use cases:
- When the play intention is triggered by the user
- When there is an error **before** playback is triggered in order to properly collect it
- When the customer wants to track some custom events via public API of `ConvivaAnalytics`
  - `sendCustomApplicationEvent`
  - `sendCustomPlaybackEvent` 
- When the customer explicitly wants to start the session via `initializeSession`

IMO and IIRC this was already covered before so I wonder why this new flag would be needed 🤔 

When you look at the `onError` or `onPlay` method listener this is already implemented in that way. 

---

The changes look to me like there changed something when it comes to the `error` use case as there you pass `fromPlay: false` which will result in not creating a new session which might lead to issues when `play` never happened before 🤔 

---

Happy to discuss this offline (slack) for faster turnaround time if needed 🙂